docs/articles/documentation/guides/advanced-guides/multiple-browser-windows.md

```diff
@@ -0,0 +1,138 @@
+---
+layout: docs
+title: Multiple Browser Windows
+permalink: /documentation/guides/advanced-guides/multiple-browser-windows.html
+---
+# Multiple Browser Windows
+
+The TestCafe API includes methods that open, close, and switch between browser windows. You can test websites with pop-up windows and OAuth login forms, debug complex multi-window applications, or run multiple instances of the same web app side by side.
+
+⚠ This is a **beta** feature. Not recommended for use in production environments. Browser support is limited to local instances of Chrome and Firefox. Videos and screenshots of child windows cannot be captured. The available functionality is subject to further revisions.
+
+️🛠️ Use the `--disable-multiple-windows` CLI flag to disable this feature if you encounter compatibility issues with legacy tests.
+
+## Handle client-side window events
+
+When your page launches a new window, the test automatically continues in the newly open window. When that window is closed, the test switches back to its parent.
+
+```js
+import { Selector } from 'testcafe';
+
+fixture `Login page`
+    .page('https://login.wrike.com/login/');
+
+const googleButton = Selector('div.login-panel-footer__login-with > button');
+
+test('Login via Google', async t => {
+    await t
+        .click(googleButton)
+        .typeText('input[type=email]', 'This text will be entered inside the login dialog');
+});
+```
+
+## Open a new window
+
+Use the [t.openWindow](../../reference/test-api/testcontroller/openwindow.md) method to open a new window:
+
+```js
+import { Selector, ClientFunction } from 'testcafe';
+
+fixture `Test page`
+    .page('https://devexpress.github.io/testcafe/example/');
+
+test('Open a new window', async t => {
+    await t.openWindow('http://example.com');
+
+    const url = await t.eval(() => document.documentURI);
+    await t.expect(url).eql('http://example.com/');
+});
+```
+
+>All open browser windows share a single pool of client-side storage. Only one [user role](https://devexpress.github.io/testcafe/documentation/guides/advanced-guides/authentication.html#user-roles) can be active at a time.
+
+## Switch between windows
+
+The [t.switchToWindow](../../reference/test-api/testcontroller/switchtowindow.md) method lets you switch between browser windows.
+
+It takes an instance of the Window object as an argument:
```

Comment:
So what was our final decision regarding "window object"? Was it "window descriptor"?

I'm okay with that, although "window identifier" seems slightly better to me.

Reply:
There can also be no "instances of objects" BTW. You can create a class instance in some languages, but here we should just say "object" (identifier/descriptor).